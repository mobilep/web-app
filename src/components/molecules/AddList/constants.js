export const keyMap = {
	ENTER: 13,
	UP_ARROW: 38,
	DOWN_ARROW: 40,
	TAB: 9,
};

export default {
	keyMap,
};
