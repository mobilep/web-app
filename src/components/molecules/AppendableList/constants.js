export const LIST_TYPES = {
	CHECK: 'check',
	STEP: 'step',
};

export default {
	LIST_TYPES,
};
