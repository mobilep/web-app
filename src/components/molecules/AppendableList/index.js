export { default } from './AppendableList';
export { default as constants } from './constants';
