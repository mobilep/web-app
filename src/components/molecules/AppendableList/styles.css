.AppendableList {
	padding: 0;
	margin: 0;
	overflow-y: auto;
	max-height: 500px;
}

.AppendableList-listItem {
	margin-bottom: 6px;
}
