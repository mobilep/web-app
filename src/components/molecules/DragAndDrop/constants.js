export const DRAG_START_EVENTS = ['dragover', 'dragenter', 'dragstart'];
export const DRAG_END_EVENTS = ['dragleave', 'dragend', 'drop'];
