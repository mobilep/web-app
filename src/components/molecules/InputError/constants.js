export const POSITIONS = {
	TOP_LEFT_POP_UP: 'topLeftPopUp',
	TOP_RIGHT_POP_UP: 'topRightPopUp',
	BOTTOM_RIGHT_POP_UP: 'bottomRightPopUp',
	BOTTOM_LEFT_POP_UP: 'bottomLeftPopUp',
	RIGHT_POP_UP: 'rightPopUp',
	LEFT_POP_UP: 'leftPopUp',
};

export const COLOR = {
	RED_ICON: 'red',
	WHITE_ICON: 'white',
};

export default {
	POSITIONS,
	COLOR,
};
