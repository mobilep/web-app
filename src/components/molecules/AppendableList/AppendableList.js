import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { AddItem } from '../../atoms';

import AppendableListItem, { constants } from '../../atoms/AppendableListItem';

import './styles.css';

import { LIST_TYPES } from './constants';
const { TYPES } = constants;

const TYPE_MAP = {
	[LIST_TYPES.CHECK]: TYPES.CHECK,
	[LIST_TYPES.STEP]: TYPES.STEP,
};

const CLASSNAME = 'AppendableList-listItem';

export default class AppendableList extends Component {

	static propTypes = {
		editable: PropTypes.bool,
		items: PropTypes.array,
		onAddItem: PropTypes.func,
		onChangeItem: PropTypes.func,
		onRemoveItem: PropTypes.func,
		placeholder: PropTypes.string,
		type: PropTypes.oneOf(Object.values(LIST_TYPES)),
		withInput: PropTypes.bool,
	};

	state = {
		inputValue: '',
	}
	generateAddButton = ({ items, onAddItem }) => (
		<AppendableListItem
			className={CLASSNAME}
			index={items.length + 1}
			uid='add-button'
			type={TYPES.ADD}
			onClick={onAddItem}
		>
			Add a few steps...
		</AppendableListItem>
	);

	generateListItem = (content, index) => (
		<AppendableListItem {...{
			editable: this.props.editable,
			onRemoveItem: this.handleRemoveItem(index),
			className: CLASSNAME,
			key: index,
			index,
			type: TYPE_MAP[this.props.type],
		}}>
			{content}
		</AppendableListItem>
	);

	handleRemoveItem = (index) =>
		() => this.props.onRemoveItem(index);

	handleAddItem = (itemValue) => {
		this.props.onAddItem(itemValue);
	}

	render () {
		const { editable, withInput = true, items = [], placeholder = 'Add item', onChangeItem } = this.props;
		return (
			<ul className='AppendableList'>
				{items.map(this.generateListItem)}
				{editable && withInput
					? <AddItem
						maxLength={128}
						placeholder={placeholder}
						onNewItem={this.handleAddItem}
						onChange={onChangeItem}
					/>
					: null
				}
			</ul>
		);
	}

}
