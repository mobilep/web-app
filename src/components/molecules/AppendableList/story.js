import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { selectV2 } from '@storybook/addon-knobs';
import { withState } from '@dump247/storybook-state';
import { titleCaseKeys } from '../../../utils';

import AppendableList from './AppendableList';

import { LIST_TYPES } from './constants';

const CASED_TYPES = titleCaseKeys(LIST_TYPES);
const DEFAULT_ITEMS = [
	'Some item',
	'Another item',
	'Third item',
];

const DEFAULT_STATE = {
	items: DEFAULT_ITEMS,
};

storiesOf('2. Molecules / AppendableList', module)
	.add('with static list', () => (
		<AppendableList
			items={DEFAULT_ITEMS}
			onAddItem={action('Clicked')}
			type={selectV2('List type', CASED_TYPES, LIST_TYPES.CHECK)}
		/>
	))
	.add('with editable list', withState(DEFAULT_STATE)(({ store }) => (
		<AppendableList
			{...store.state}
			onRemoveItem={(index) => store.set({ items: store.state.items.filter((item, i) => index !== i) }) }
			onAddItem={(item) => store.set({ items: store.state.items.concat([item]) })}
			type={selectV2('List type', CASED_TYPES, LIST_TYPES.CHECK)}
			editable
		/>
	)));
