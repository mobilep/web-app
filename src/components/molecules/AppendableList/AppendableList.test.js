import React from 'react';
import { shallow } from 'enzyme';

import AppendableList from './AppendableList';
import { LIST_TYPES } from './constants';

const ITEMS = [
	'Some item',
	'Another item',
	'Third item',
];
const TYPES_ARRAY = Object.values(LIST_TYPES);

describe('Molecule - AppendableList', () => {

	TYPES_ARRAY.forEach((type) => {
		it(`Should render with type ${type}`, () => {
			const wrapper = shallow(
				<AppendableList
					items={ITEMS}
					type={type}
				/>
			);
			expect(wrapper).toMatchSnapshot();
		});
	});

	it('Should render as editable', () => {
		const wrapper = shallow(
			<AppendableList
				items={ITEMS}
				type={LIST_TYPES.CHECK}
				editable
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});

});
