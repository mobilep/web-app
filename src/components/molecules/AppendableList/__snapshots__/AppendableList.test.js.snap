// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecule - AppendableList Should render as editable 1`] = `
<ul
  className="AppendableList"
>
  <AppendableListItem
    className="AppendableList-listItem"
    editable={true}
    index={0}
    key="0"
    onRemoveItem={[Function]}
    type="check"
  >
    Some item
  </AppendableListItem>
  <AppendableListItem
    className="AppendableList-listItem"
    editable={true}
    index={1}
    key="1"
    onRemoveItem={[Function]}
    type="check"
  >
    Another item
  </AppendableListItem>
  <AppendableListItem
    className="AppendableList-listItem"
    editable={true}
    index={2}
    key="2"
    onRemoveItem={[Function]}
    type="check"
  >
    Third item
  </AppendableListItem>
  <AddItem
    maxLength={128}
    onNewItem={[Function]}
    placeholder="Add item"
  />
</ul>
`;

exports[`Molecule - AppendableList Should render with type check 1`] = `
<ul
  className="AppendableList"
>
  <AppendableListItem
    className="AppendableList-listItem"
    index={0}
    key="0"
    onRemoveItem={[Function]}
    type="check"
  >
    Some item
  </AppendableListItem>
  <AppendableListItem
    className="AppendableList-listItem"
    index={1}
    key="1"
    onRemoveItem={[Function]}
    type="check"
  >
    Another item
  </AppendableListItem>
  <AppendableListItem
    className="AppendableList-listItem"
    index={2}
    key="2"
    onRemoveItem={[Function]}
    type="check"
  >
    Third item
  </AppendableListItem>
</ul>
`;

exports[`Molecule - AppendableList Should render with type step 1`] = `
<ul
  className="AppendableList"
>
  <AppendableListItem
    className="AppendableList-listItem"
    index={0}
    key="0"
    onRemoveItem={[Function]}
    type="step"
  >
    Some item
  </AppendableListItem>
  <AppendableListItem
    className="AppendableList-listItem"
    index={1}
    key="1"
    onRemoveItem={[Function]}
    type="step"
  >
    Another item
  </AppendableListItem>
  <AppendableListItem
    className="AppendableList-listItem"
    index={2}
    key="2"
    onRemoveItem={[Function]}
    type="step"
  >
    Third item
  </AppendableListItem>
</ul>
`;
